modules/percolator/src/main/java/org/elasticsearch/percolator/PercolatorMatchedSlotSubFetchPhase.java

```diff
@@ -71,11 +74,8 @@ static void innerHitsExecute(Query mainQuery, IndexSearcher indexSearcher, Searc
         for (PercolateQuery percolateQuery : percolateQueries) {
             String fieldName = singlePercolateQuery ? FIELD_NAME_PREFIX : FIELD_NAME_PREFIX + "_" + percolateQuery.getName();
             IndexSearcher percolatorIndexSearcher = percolateQuery.getPercolatorIndexSearcher();
-            // there is a bug in lucene's MemoryIndex that doesn't allow us to use docValues here...
-            // See https://issues.apache.org/jira/browse/LUCENE-8055
-            // for now we just use version 6.0 version to find nested parent
-            final Version version = Version.V_6_0_0; //context.mapperService().getIndexSettings().getIndexVersionCreated();
-            Weight weight = percolatorIndexSearcher.createWeight(percolatorIndexSearcher.rewrite(Queries.newNonNestedFilter(version)),
+            Query nonNestedQuery = Queries.newNonNestedFilter(indexVersionCreated);
```

Comment:
We can remove the version here too since it only checks if the index was created before 6.1.0. We can do this in a follow up but we probably don't need to add the `indexVersionCreated` param here and just use CURRENT until we remove it from `newNonNestedFilter` ?

Reply:
👍 That makes sense. I will update this PR.